node/src/ledger/read/ledger_read_actions.rs

```diff
@@ -27,7 +27,9 @@ pub enum LedgerReadAction {
 impl redux::EnablingCondition<crate::State> for LedgerReadAction {
     fn is_enabled(&self, state: &crate::State, _time: redux::Timestamp) -> bool {
         match self {
-            LedgerReadAction::FindTodos => state.ledger.read.is_total_cost_under_limit(),
+            LedgerReadAction::FindTodos => {
+                state.p2p.ready().is_some() && state.ledger.read.is_total_cost_under_limit()
```

Comment:
Read requests arent just for p2p. Please revert this change.

Reply:
Hmm... ok, there was a place where p2p was kind of essential, i.e. we collected ready peers. But for uninitialized p2p it will be just empty list.